src/lib.rs

```diff
@@ -1518,6 +1517,14 @@ impl Rtc {
             return self.ice.accepts_message(v);
         }
 
+        // Slow path: Occasionally, traffic comes in on a socket address corresponding
+        // to a successful candidate pair other than the one we've currently nominated.
+        // This typically happens at the beginning of the connection
+        let remote_candidates = self.ice.remote_candidates();
+        if remote_candidates.iter().any(|cand| cand.addr() == r.source) {
+            return true;
+        }
```

Comment:
Can you make this a function on `IceAgent` to avoid having too much of the logic here in `Rtc`? Something like `has_remote_candidate`.